x/staking/keeper/msg_server.go

```diff
@@ -210,6 +235,13 @@ func (k msgServer) Delegate(goCtx context.Context, msg *types.MsgDelegate) (*typ
 		return nil, err
```

Comment:
Can we move up this check and the amount check before we try to get the validator?
And wrap the error message like it was in validate basic?